### ISSUE TITLE
scheduler-perf: add a test case to confirm QHint's impact on the scheduling throughput

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes/kubernetes#127140]
The current test cases of scheduler_perf are basically simple; create Pods with a specific template (i.e., specific scheduling constraint etc) and measure the metrics.

But, the real cluster actually often has various unschedulable Pods, and each of them has different unschedulable plugins. 
By adding such a scenario in scheduler-perf, we can observe QHint's impact on the scheduling throughput more, since QHints accurately selects Pods to move to activeQ/backoffQ based on each Pod's unschedulable reason.

/sig scheduling
/kind feature

**Comment by sanposhiho** [comment]
/cc @macsko 

I just filed what I talked about in DM so that we don't forget.

**Comment by chansuke** [comment]
may I try on this issue?

**Comment by sanposhiho** [comment]
@macsko you wanna have it on your plate? or ok to be taken over?

**Comment by macsko** [comment]
I'm focusing on writing the tests utilizing `scheduler_queueing_hint_execution_duration_seconds` and `scheduler_event_handling_duration_seconds` metrics, so I can hand this issue over.

**Comment by sanposhiho** [comment]
@chansuke Any update? Do you want someone else to take over?

**Comment by chansuke** [comment]
@sanposhiho 
oh..no update and i'd be happy for someone  to take it over. thank you!

**Comment by sanposhiho** [comment]
Sure, unassign you for now.

/unassign

Feel free to pick it up again in case you get an enough time.

**Comment by nayihz** [comment]
I find that there are already test case which creates unscheduled pod. Does it not meet the requirements? Could you elaborate further on what we needl? Thanks.
https://github.com/kubernetes/kubernetes/blob/2d0a4f75560154454682b193b42813159b20f284/test/integration/scheduler_perf/event_handling/performance-config.yaml#L412-L420

**Comment by sanposhiho** [comment]
Each test case that @macsko has worked on is designed to measure the performance in a very specific scenario.

For example, `EventHandlingPodAdd` test case that you showed does:
- all the pods are rejected by the plugins that registers Pod/Add event. 
- `createPods` to unblock those pods.

So, in this test case, the scheduler only gets Pod/Add event, and basically all pods in the unschedQ are retried with all those events. But, in a real world cluster, the scheduler always gets many different kinds of events, both which make pods schedulable and which don't make pods schedulable; and even within Pod/Add events, there's some that don't make unsched pods schedulable, and some that make them schedulable. 
And, that's where QHint would show its power more; QHint properly evaluates which events really likely make each Pod schedulable and makes a decision to reduce unnecessary requeueing.

So, we need a scheduler-perf scenario that observes the performance within a more chaotic scenario; making different kinds of pods that would be rejected by different plugins and then making different kinds of cluster events, which each would requeue some pods (not all pods). That's what this issue wants to add.

**Comment by nayihz** [comment]
> making different kinds of pods that would be rejected by different plugins and then making different kinds of cluster events, which each would requeue some pods (not all pods).

I get your point. But should we implement it in **one** test case? There are too many kinds of cluster events so I think this is a bit difficult to cover all the scenarios in one test case.

**Comment by sanposhiho** [comment]
Yes, that's true. Either one huge test or separate tests is fine to me as long as those satisfy my intention described above.

**Comment by saku3** [comment]
may I try on this issue?

**Comment by sanposhiho** [comment]
Yes,

/assign @saku3
